src/parse.rs

```diff
@@ -53,36 +41,21 @@ fn parse_a_hrefs(handle: &Handle, root_url: &Url, base_url: &Url, urls: &mut Has
                 } else {
                     debug!("unparsable link {:?}", href);
                 }
-            }
-        }
-    }
+                Ok(())
+            })],
+            ..RewriteStrSettings::default()
+        },
+    )
+    .expect("html rewriting failed");
 
-    for child in node.children.borrow().iter() {
-        parse_a_hrefs(&child, root_url, base_url, urls);
-    }
+    urls
 }
 
 #[cfg(test)]
 mod test {
-    use html5ever::parse_document;
-    use html5ever::{rcdom::RcDom, tendril::TendrilSink};
-    use std::collections::HashSet;
+    use super::parse_a_hrefs as gather_urls;
```

Comment:
Nitpick: perhaps use it as `parse_a_hrefs` and refactor `gather_urls` to `parse_a_hrefs` below?